File: Npc/11003335.cs
using Maple2.Model.Enum;
using Maple2.Script.Npc;

/// <summary>
/// 11003335: Barista
/// </summary>
public class _11003335 : NpcScript {
    protected override int First() {
        return 10;
    }

    // Select 0:
    // $script:0510143807008459$
    // - What do you want?!
    protected override int Select() => 0;

    protected override int Execute(int selection) {
        switch (Id, Index++) {
            case (10, 0):
                // $script:0510145607008463$
                // - You'll get nothing out of me!
                return -1;
        }
        
        return default;
    }

    protected override NpcTalkButton Button() {
        return (Id, Index) switch {
            (10, 0) => NpcTalkButton.Close,
            _ => NpcTalkButton.None,
        };
    }
}

File: Npc/11003335.py
""" 11003335: Barista """
from npc_api import Option, Script
import random


class Main(Script):
    def first(self) -> int:
        return 10

    def select(self) -> int:
        return 0

    def __0(self, pick: int) -> int:
        # $script:0510143807008459$
        # - What do you want?!
        return None # TODO

    def __10(self, pick: int) -> int:
        # $script:0510145607008463$
        # - You'll get nothing out of me!
        return -1

    def button(self) -> Option:
        if (self.state, self.index) == (10, 0):
            return Option.CLOSE
        return Option.NONE